src/index.ts

```diff
@@ -0,0 +1,52 @@
+import * as  amqp from 'amqplib/callback_api';
+import { Channel, Connection, Message } from 'amqplib/callback_api';
+import { config } from 'dotenv'; config(); //initialize right there the configuration
+import { readFileSync } from 'fs';
+import { QueueMessage } from './types';
+import { PayloadSchema } from './schemas';
+import { connectToDatabase } from './datasource';
+import { Logger } from './utils';
+import * as Joi from 'joi';
+import {
+  recordStartMigration,
+  getClientId,
+  recordStructureValidationStatus,
+}
+  from './modules';
+
+amqp.connect(process.env.AVW_QUEUE_HOST || 'amqp://localhost', function (error: Error, connection: Connection) {
+  connection.createChannel(function (error: Error, channel: Channel) {
+    var queueName = process.env.AVW_QUEUE_NAME || "DHIS2_VALIDATION_QUEUE";
+    channel.assertQueue(queueName, { durable: true });
+    console.log("[*] Waiting for messages on %s. To exit press CTRL+C", queueName);
+    channel.consume(queueName, async function (message: Message | null) {
+      if (message) {
+        const queueMessage: QueueMessage = JSON.parse(message.content.toString());
+        const logger = new Logger(queueMessage.channelId);
+        const sequelize = await connectToDatabase(
+          process.env.AVW_DATABASE_HOST || 'localhost',
+          process.env.AVW_DATABASE || '',
+          process.env.AVW_DATABASE_USERNAME || 'username',
+          process.env.AVW_DATABASE_PASSWORD || ''
+        );
+        const clientId = await getClientId(sequelize, queueMessage.clientId);
+
+        const migration = await recordStartMigration(sequelize, clientId);
+
+        const payloadFile = `${process.env.AVW_PAYLOADS_ROOT_DIR}/${queueMessage.channelId}.adx`;
+        const payload = JSON.parse(readFileSync(payloadFile).toString());
```

Comment:
what happens if we can't open the file, or it doesn't exist? we should probably handle this

Reply:
Handled now